api/firstbook.py

```diff
@@ -40,26 +41,26 @@ class FirstBookAuthenticationAPI(BasicAuthenticationProvider):
     DEFAULT_PASSWORD_REGULAR_EXPRESSION = '^[0-9]+$'
 
     SETTINGS = [
-        { "key": ExternalIntegration.URL, "format": "url", "label": _("URL"), "required": True },
-        { "key": ExternalIntegration.PASSWORD, "label": _("Key"), "required": True },
+        {"key": ExternalIntegration.URL, "format": "url",
+            "label": _("URL"), "required": True},
+        {"key": ExternalIntegration.PASSWORD,
+            "label": _("Key"), "required": True},
     ] + BasicAuthenticationProvider.SETTINGS
 
     log = logging.getLogger("First Book authentication API")
 
-    def __init__(self, library_id, integration, analytics=None, root=None):
-        super(FirstBookAuthenticationAPI, self).__init__(library_id, integration, analytics)
+    def __init__(self, library_id, integration, analytics=None, root=None, secret=None):
+        super(FirstBookAuthenticationAPI, self).__init__(
+            library_id, integration, analytics)
         if not root:
-            url = integration.url
-            key = integration.password
-            if not (url and key):
+            root = integration.url
+            self.key = secret or integration.password
+            if not (root and self.key):
                 raise CannotLoadConfiguration(
                     "First Book server not configured."
                 )
-            if '?' in url:
-                url += '&'
-            else:
-                url += '?'
-            root = url + 'key=' + key
+        if root[-1] == '/':
+            root = root[:-1]
```

Comment:
I see the utility in this check, what prompted its inclusion though? We should be able to trust the URL stored in the database.

Reply:
The original firstbook auth seemed to require significant url manipulation.  I added that for a test that was failing in the original test suite.  The tests have been updated as this version does not require this. And this was removed.